examples/adaptiveprecision-blockjacobi/CMakeLists.txt

```diff
@@ -1,5 +1,12 @@
-set(target_name "adaptiveprecision-blockjacobi")
-add_executable(${target_name} ${target_name}.cpp)
-target_link_libraries(${target_name} Ginkgo::ginkgo)
-target_include_directories(${target_name} PRIVATE ${PROJECT_SOURCE_DIR})
+cmake_minimum_required(VERSION 3.9)
+project(adaptiveprecision-blockjacobi)
+
+# We only need to find Ginkgo if we build this example stand-alone
+if (NOT GINKGO_BUILD_EXAMPLES)
+    find_package(Ginkgo 1.4.0 REQUIRED)
```

Comment:
Is there a need to require a specific version ?

Reply:
This means that we find Ginkgo with version number at least 1.4.0 and the same major version (i.e. no 2.0), so we can be sure that the interfaces we use are available.

Reply:
Yes, I mean do we need 1.4.0 ? 1.3.0 should also work ? Because, technically these are standalone and can be used with an already existing, possibly older version of Ginkgo ?

Reply:
That is a good point, in general I agree. However, with the addition of `DpcppExecutor` to most examples, they are no longer backwards-compatible with 1.3.0, so we need to modify it in a few places at least.

Reply:
For ease of maintainance and version bumps, can you create a shared `.cmake` file somewhere to define a variable like ${GINKGO_VERSION_REQUIRED}` or whatever the name could be? Just so that it's in one place.

You would also need to set that file as installed or it can be to the directory when building the example.

Reply:
My idea behind these changes is to make the CMake files as self-sufficient as possible. So I would like to avoid such external dependencies. Another point is that this way, you only need to increment the version requirement if you actually modify the example to use newer parts of Ginkgo, but whether that is useful would be another question entirely.

Reply:
My fear is that this will easily become hard to manage and some aspects might be missed which make examples actually incompatible but we did not realize that. For that reason, I think I would pretty much always bump the version requirement if it is there.

Reply:
I see. So would you be okay with keeping the values in sync manually, but leaving them in here explicitly?

Reply:
I think that's still somewhat error prone particularly as the number of examples increase, but if you think there is no easy way to integrate a macro in that workflow, fine then.